tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php

```diff
@@ -85,4 +85,255 @@ public function removeInvisibleNodesRemovesNodesWithDisplayNone($displayNone)
 
         self::assertNotContains('<div', $subject->render());
     }
+
+    /**
+     * @test
+     */
+    public function removeRedundantClassesProvidesFluentInterface()
+    {
+        $subject = HtmlPruner::fromHtml('<html></html>');
+
+        $result = $subject->removeRedundantClasses([]);
+
+        self::assertSame($subject, $result);
+    }
+
+    /**
+     * @return string[][][]
+     */
+    public function classesToKeepDataProvider()
+    {
+        return [
+            'no classes to keep' => [[]],
+            '1 class to keep' => [['foo']],
```

Comment:
To cover all equivalence classes, we'll also need a test case for 2 classes to keep (for the minimum version of "multiple classes").

Reply:
OK, will add it.